providers/policy.go

```diff
@@ -0,0 +1,20 @@
+/*
+ Copyright 2024 The OSCAL Compass Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package providers
+
+import "github.com/oscal-compass/oscal-sdk-go/extensions"
+
+/*
+PolicyProvider defines methods for a policy validation engine.
+Defined uses cases include the following:
+1. A scanning plugin may contact a remote API for scanning
+2. A scanning plugin may exec out to another tool for scanning in a new process
+3. A scanning plugin may be a self-contained scanning tool
+*/
+type PolicyProvider interface {
```

Comment:
```suggestion
type Policy interface {
```
This is just a nit, but it would be imported in other packages as `providers.Policy` with the change.